src/mixins/java/org/spongepowered/common/mixin/core/server/level/ServerPlayerMixin.java

```diff
@@ -903,6 +904,19 @@ public void sendMessage(final net.minecraft.network.chat.Component p_241151_1_,
             }
         }
     }
+
+    @Inject(method = "setPlayerInput", at = @At("HEAD"))
+    public void onSetPlayerInput(final float sway, final float surge, final boolean jumping, final boolean dismount, final CallbackInfo ci) {
+        SpongeCommon.postEvent(SpongeEventFactory.createSteerVehicleEvent(
```

Comment:
> This is going to fire if the Player Input Packet is sent, regardless of whether the player is actually mounted. It would make more sense to redirect isPassenger and fire there if isPassenger would return true.

That's probably a good idea :sweat_smile: 

> Is it because the client dismounts without validation from the server?

That's exactly what I meant! Sorry for being unclear.